source/builders.txt

```diff
@@ -0,0 +1,305 @@
+
+The following steps describe how to create an update document by using
+builders:
+
+1. Create a ``Pipeline`` instance.
+   
+#. Pass one or more ``$set`` stages by calling the ``Stage::set()``
+   method and passing a field name and a value.
+
+The following code shows the template to define an update by using
+builders:
+
+.. code-block:: php
+    
+   $update = new Pipeline(
+       Stage::set(<field name>: <value>),
+       Stage::set(<field name>: <value>),
+       ...
+   );
+
+This example performs the following actions:
+
+- Uses the ``Query::eq()`` factory method to match documents in which
+  the ``watlev`` field value is ``'partly submerged at high water'``
+- Uses builders to specify an update that sets
+  sets the ``year`` field to ``1870``
+- Calls the :phpmethod:`MongoDB\Collection::updateOne()`
+  method to perform the update
+- Prints the number of updated documents
+
+.. io-code-block::
+   :copyable:   
+
+   .. input:: /includes/builders.php
+      :start-after: start-updateone
+      :end-before: end-updateone
+      :language: php
+      :dedent:
+
+   .. output::
+      :language: none
+      :visible: false
+
+      Updated documents: 1
+
+.. _php-builders-changestream:
+
+Modify Change Stream Output
+---------------------------
+
+You can use factory methods from the ``Stage`` class to modify a change
+stream's output by creating a pipeline. To learn more about change
+streams, see the :ref:`php-change-streams` guide.
+
+The following steps describe how to create a change stream filter by
+using builders:
+
+1. Create an array.
+   
+#. Pass one or more ``$match`` stages by calling factory methods from
+   the ``Stage`` class and the required parameters.
+
+The following code shows the template to modify change stream output by
+using builders:
+
+.. code-block:: php
+
+   $pipeline = [
+       Stage::<factory method>(...),
+       Stage::<factory method>(...),
+       ...
+   ];
+
+You can pass this pipeline to the following methods:
+
+- :phpmethod:`MongoDB\Client::watch()`
+- :phpmethod:`MongoDB\Database::watch()`
+- :phpmethod:`MongoDB\Collection::watch()`
+
+This example performs the following actions:
+
+- Uses the ``Stage::match()`` method to filter for only update change
+  events
+- Uses the ``Stage::project()`` method to output only the specified change
+  event fields
+- Calls the :phpmethod:`MongoDB\Collection::watch()`
+  method to open the change stream with the option to output the full
+  document after update
+- Prints change events as they occur
+
+.. io-code-block::
+   :copyable:   
+
+   .. input:: /includes/builders.php
+      :start-after: start-cs
+      :end-before: end-cs
+      :language: php
+      :dedent:
+
+   .. output::
+      :language: json
+      :visible: false
+
+      {
+        "_id":...,
+        "operationType":"update",
+        "fullDocument":{"_id":...,"feature_type":"Wrecks - Visible",...},
+        "ns":{"db":"sample_geospatial","coll":"shipwrecks"}
+      }
```

Comment:
S: And API documentation section would be helpful 
```suggestion
      }
     
Additional Information
----------------------
```

Reply:
see other reply about API documentation